contracts/integrations/test/utils/cache.ts

```diff
@@ -0,0 +1,90 @@
+import { ContractGetterFunction } from './function_assertions';
+import * as _ from 'lodash';
+
+export class GetterCache {
+    // The getter function whose values will be cached.
+    public getter: ContractGetterFunction;
+
+    // The cache that will be used to store values. This has to use a "string" for indexing
+    // because the "Map" datastructure uses reference equality when the keys are objects,
+    // which was unsuitable for our purposes.
+    private cache: {
+        [key: string]: any;
+    };
+
+    constructor(getter: ContractGetterFunction) {
+        this.getter = getter;
+        this.cache = {};
+    }
+
+    /**
+     * Calls the contract getter and caches the result if there is not a cached value. Otherwise,
+     * this returns the cached value.
+     * @param ...args A variadic list of args to use when calling "callAsync". Due
+     *                to the fact that we need to serialize the arguments to "callAsync" these
+     *                arguments must be valid arguments to "getABIEncodedTransactionData".
+     * @return Either a cached value or the queried value.
+     */
+    public async callAsync(...args: any[]): Promise<any> {
+        const key = this.getter.getABIEncodedTransactionData(...args);
+        const cachedResult = this.cache[key];
+
+        if (cachedResult !== undefined) {
+            return cachedResult;
```

Comment:
optional: I feel like there should be a way to update the cache without manually flushing it?

Reply:
I agree with this sentiment. As discussed offline, the getter cache was originally intended to fulfill a pretty specific use case (it is intended to cache values between `before` or `after` calls). In this specific case, this type of cache works well; however, a different mechanism would also be good. 

One idea that I have is to actually include the getter cache at a lower level of abstraction in our tools (probably in something like a contract wrappers). In this model, the getter caches could be flushed whenever a relevant stateful function is called. This could potentially use the `GetterCache` implementation from this PR, but it would effectively encapsulate when the cache needs to be flushed.

Reply:
I feel like it'd be fine to go with a simple solution (e.g. a flag for `callAsync` to update the cache) until we understand the use cases for this cache better

Reply:
After discussing offline with @dorothy-zbornak, I'm going to remove the `GetterCache` for now since it isn't being used at all. We can add this back in later when we have a use for it (I'm going to keep the commit hash around), by which point I expect that we'll know the right way to flush the cache ergonomically.